nuget 5.9 pack crashes

## Details about Problem

NuGet product used (NuGet.exe):

Product version: 5.9.0

Worked before? If so, with which NuGet version: 5.8.1

## Repro steps and/or sample project

Here is the error caused in 5.9.0:

https://github.com/cryptlex/lexactivator-dotnet/runs/2181577286?check_suite_focus=true 

in Pack Nuget Package step.

and it worked till 5.8.1

https://github.com/cryptlex/lexactivator-dotnet/runs/1976606695?check_suite_focus=true 

### Verbose Logs
```
nuget pack src\Cryptlex.LexActivator\Cryptlex.LexActivator.nuspec -NonInteractive -OutputDirectory nupkg -Properties Configuration=Release -Verbosity Detailed
NuGet Version: 5.9.0.7134
Attempting to build package from 'Cryptlex.LexActivator.nuspec'.
Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at NuGet.Packaging.PackageBuilder.<>c.<ValidateReferenceAssemblies>b__173_1(NuGet.Frameworks1204688.NuGetFramework groupFramework)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
   at NuGet.Packaging.PackageBuilder.ValidateReferenceAssemblies(IEnumerable`1 files, IEnumerable`1 packageAssemblyReferences)
   at NuGet.Packaging.PackageBuilder.Save(Stream stream)
   at NuGet.Commands.PackCommandRunner.BuildPackage(PackageBuilder builder, String outputPath, Boolean symbolsPackage)
   at NuGet.Commands.PackCommandRunner.BuildFromNuspec(String path)
   at NuGet.CommandLine.PackCommand.ExecuteCommand()
   at NuGet.CommandLine.Command.ExecuteCommandAsync()
   at NuGet.CommandLine.Command.Execute()
   at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args)
Error: Process completed with exit code 1.
```

--- a/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs
+++ b/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs
@@ -606,7 +606,7 @@ namespace NuGet.Packaging
         {
             var frameworksMissingPlatformVersion = new HashSet<string>(packageAssemblyReferences
                 .Select(group => group.TargetFramework)
-                .Where(groupFramework => groupFramework.HasPlatform && groupFramework.PlatformVersion == FrameworkConstants.EmptyVersion)
+                .Where(groupFramework => groupFramework != null && groupFramework.HasPlatform && groupFramework.PlatformVersion == FrameworkConstants.EmptyVersion)
                 .Select(framework => framework.GetShortFolderName()));
             if (frameworksMissingPlatformVersion.Any())
             {

--- a/test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs
+++ b/test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs
@@ -2292,6 +2292,24 @@ Description is required.");
         }
 
         [Fact]
+        public void ValidateReferencesAllowsNullFramework()
+        {
+            // Arrange
+            var files = new[] {
+                        new PhysicalPackageFile { TargetPath = @"lib" + Path.DirectorySeparatorChar + "net40" + Path.DirectorySeparatorChar + "foo.dll" },
+                        new PhysicalPackageFile { TargetPath = @"lib" + Path.DirectorySeparatorChar + "net40" + Path.DirectorySeparatorChar + "bar.dll" },
+                        new PhysicalPackageFile { TargetPath = @"lib" + Path.DirectorySeparatorChar + "net40" + Path.DirectorySeparatorChar + "baz.exe" },
+                    };
+            var packageAssemblyReferences = new PackageReferenceSet((NuGetFramework)null, new string[] { "foo.dll", "bar", "baz" });
+
+            // Act and Assert
+            PackageBuilder.ValidateReferenceAssemblies(files, new[] { packageAssemblyReferences });
+
+            // If we've got this far, no exceptions were thrown.
+            Assert.True(true);
+        }
+
+        [Fact]
         public void ValidateReferencesThrowsForPartialNamesThatDoNotHaveAKnownExtension()
         {
             // Arrange

